application/controllers/ServicesController.php

```diff
@@ -199,6 +203,95 @@ public function searchEditorAction()
         $this->setTitle(t('Adjust Filter'));
     }
 
+    public function gridAction()
+    {
+        $db = $this->getDb();
+        $this->setTitle(t('Service Grid'));
+
+        $query = Service::on($db)->with([
+            'state',
+            'host',
+            'host.state'
+        ]);
+
+        $gridcols = [
+            'host_name' => 'host.display_name',
+            'host_display_name' => 'host.name',
+            'service_name' => 'service.name',
+            'service_display_name' => 'display_name',
+            'service_handled' => 'service.state.is_handled',
+            'service_output' => 'service.state.output',
+            'service.state.soft_state'
```

Comment:
It is easier to get axis keys this way using $row->{$this->yAxisColumn} or $row->{$this->xAxisColumn} when the axis column is 'host.name'.

Reply:
I see. Why is `service.state.soft_state` not aliased then?

Reply:
I did not use alias there because in `grid.phtml` it would be easier to use `$service->state->getStateText()`. But now I have changed it. I have given alias even for `service.state.soft_state`.